reporting/slack.go

```diff
@@ -27,150 +32,216 @@ func (s *SlackReporter) buildSummaryReport(
+	summaryReport := s.buildSummaryReport(header, numRepos, report, reportTime)
 	wg.Add(1)
-	go s.sendSlackMessage(s.config.Default_slack_channel, summaryReport, wg)
+	go s.sendSlackMessage(s.Config.Default_slack_channel, slack.MsgOptionBlocks(summaryReport.Blocks.BlockSet...), wg)
 	return nil
 }
 
-func (s *SlackReporter) buildTeamReports(
-	teamReports map[string]map[string]VulnerabilityReport,
-	reportTime string,
-) map[string]string {
-	log := logger.Get()
-	slackMessages := map[string]string{}
-
+func (s *SlackReporter) buildTeamRepositoryReport(
+	repoName string,
+	repoReport VulnerabilityReport,
+) *slack.SectionBlock {
+	var err error
+	var severityIcon string
 	severities := getSeverityReportOrder()
-	for team, repos := range teamReports {
-		teamReport := ""
-		teamInfo, err := config.GetTeamConfigBySlug(team, s.config.Team)
+	vulnCounts := make([]string, 0)
+	for _, severity := range severities {
+		if severityIcon == "" && repoReport.VulnsBySeverity[severity] > 0 {
+			severityIcon, err = config.GetIconForSeverity(severity, s.Config.Severity)
+			if err != nil {
+				severityIcon = DEFAULT_SLACK_ICON
+			}
+		}
+		vulnCounts = append(vulnCounts, fmt.Sprintf("%2d %s", repoReport.VulnsBySeverity[severity], severity))
+	}
+	if severityIcon == "" {
+		severityIcon, err = config.GetIconForSeverity("None", s.Config.Severity)
 		if err != nil {
-			log.Warn().Str("team", team).Msg("Skipping report for unconfigured team.")
-			continue
+			severityIcon = DEFAULT_SLACK_ICON
 		}
-		// Retrieve the list of repo names so that we can report alphabetically
-		repoNames := maps.Keys(repos)
-		sort.Strings(repoNames)
-		for _, name := range repoNames {
-			repo := repos[name]
-			if name == SUMMARY_KEY {
-				continue
-			}
-			repoReport := fmt.Sprintf("*%s* -- ", name)
-			for _, severity := range severities {
-				count := repo.VulnsBySeverity[severity]
-				icon, err := config.GetIconForSeverity(severity, s.config.Severity)
-				if err != nil {
-					icon = DEFAULT_SLACK_ICON
-				}
-				repoReport += fmt.Sprintf("%s *%s*: %d ", icon, severity, count)
-			}
-			repoReport += "\n"
-			teamReport += repoReport
+	}
+	fields := []*slack.TextBlockObject{
+		slack.NewTextBlockObject(slack.MarkdownType, fmt.Sprintf("%s *%s*", severityIcon, repoName), false, false),
+		slack.NewTextBlockObject(slack.MarkdownType, strings.Join(vulnCounts, " | "), false, false),
+	}
+	return slack.NewSectionBlock(nil, fields, nil)
+}
+
+func (s *SlackReporter) buildTeamReport(
+	teamID string,
+	repos map[string]VulnerabilityReport,
+	reportTime string,
+) *SlackReport {
+	log := logger.Get()
+	teamInfo, err := config.GetTeamConfigBySlug(teamID, s.Config.Team)
+	if err != nil {
+		log.Warn().Err(err).Str("team", teamID).Msg("Skipping report for unconfigured team.")
+		return nil
+	}
+	if teamInfo.Slack_channel == "" {
+		log.Debug().Str("team", teamID).Any("repos", repos).Msg("Skipping report since Slack channel is not configured.")
```

Comment:
[q] This seems like more than a Debug? Either an Info or Warn since important configuration is missing?

Reply:
I really debated back and forth on that one. The reason I went with debug is because, with a sufficient size org, this could get pretty spammy. I remember at $previousEmployer, there were thousands of repos, and that would just be... way too much.

Though I suppose at info level it wouldn't show by default 🤔 We may want to just go through and do a pass on log levels in general.